components/automate-gateway/handler/authz.go

```diff
@@ -345,17 +345,23 @@ func (a *AuthzServer) getAllowedProjects(
 	// Fetches the id of the current user PLUS the team ids for that user
 	subjects := auth_context.FromContext(ctx).Subjects
 
-	// Note that projects do not exist in v1 of course,
-	// but as project introspection is so similar to endpoint introspection,
-	// it was easy to mirror that code here.
-	resp, err := a.filterHandler.FilterAuthorizedProjects(ctx, subjects,
-		mapByResourceAndActionV1, mapByResourceAndActionV2,
-		methodsInfoV1, methodsInfoV2)
+	inputPairs := pairs.GetKeys(mapByResourceAndActionV2)
+	pairsV2 := make([]*authzV2.Pair, len(inputPairs))
+	for i, p := range inputPairs {
+		pairsV2[i] = &authzV2.Pair{Resource: p.Resource, Action: p.Action}
+	}
+
+	resp, err := a.clientV2.FilterAuthorizedProjects(ctx, &authzV2.FilterAuthorizedPairsReq{
+		Subjects: subjects,
+		Pairs:    pairsV2,
+	})
 	if err != nil {
 		log.WithError(err).Debug("Error on client.FilterAuthorizedProjects")
 		return nil, err
 	}
-	return resp.Projects, nil
+	respProjects := make([]string, len(resp.Projects))
+	copy(respProjects, resp.Projects)
+	return respProjects, nil
```

Comment:
Why do we do this `copy` business? In any real-world usage, I don't think this matters, does it?

Reply:
You are probably right; don't recall why that was here...

Reply:
I think this might have come from being consistent with some other code that was the result an overly-verbose comment I left on another PR:

https://github.com/chef/a2/pull/5396#discussion_r269438700

As I mentioned there though, I think we can return resp.Projects directly in this case.